pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -534,13 +739,47 @@ def make_parser(parser: argparse.ArgumentParser):
 
 
 def main(args):
-    path = Path(args.package[0]).resolve()
+    meta_file = Path(args.package[0]).resolve()
     if args.compress_package and not _have_terser():
         raise RuntimeError(
             "Terser is required to compress packages. Try `npm install -g terser` to install terser."
         )
 
-    build_package(path, args)
+    pkg_root = meta_file.parent
+    pkg = parse_package_config(meta_file)
+    name = pkg["package"]["name"]
+    t0 = datetime.now()
+    print("[{}] Building package {}...".format(t0.strftime("%Y-%m-%d %H:%M:%S"), name))
+    success = True
+    try:
+        pkg["source"] = pkg.get("source", {})
+        pkg["build"] = pkg.get("build", {})
+        build_metadata = pkg["build"]
+        build_metadata["cflags"] = build_metadata.get("cflags", "")
+        build_metadata["cxxflags"] = build_metadata.get("cxxflags", "")
+        build_metadata["ldflags"] = build_metadata.get("ldflags", "")
+
+        build_metadata["cflags"] += f" {args.cflags}"
+        build_metadata["cxxflags"] += f" {args.cxxflags}"
+        build_metadata["ldflags"] += f" {args.ldflags}"
```

Comment:
Was this part broken before? If so might be worth adding a changelog entry.

Reply:
No, it was fine before. An intermediate rearrangement was broken.